src/checks/series/flatline_check.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{DataCache, Error, Flag};
+
+/// Timeseries QC test that checks for streaks of repeating values.
+///
+/// If all observations passed in are identical, [`Flag::Fail`] will be returned, if any are
+/// missing, [`Flag::DataMissing`], if `data` is empty, [`Flag::Invalid`], else [`Flag::Pass`].
+pub fn flatline_check(data: &[Option<f32>]) -> Flag {
+    if data.contains(&None) {
+        return Flag::DataMissing;
+    }
+    let data: Vec<f32> = data.iter().map(|opt| opt.unwrap()).collect();
+
+    let base = match data.first() {
+        Some(base) => base,
+        None => return Flag::Invalid,
+    };
+    if !data.iter().any(|x| x != base) {
+        return Flag::Fail;
+    }
+    Flag::Pass
+}
+
+/// Apply [`flatline_check`] to a whole [`DataCache`]
+///
+/// If `num_points` observations in a row are identical, the last will be flagged as Flag::Fail, if
+/// any of the last `num_points` observations are missing, it will be flagged as Flag::DataMissing,
+/// else Flag::Pass.
+///
+/// As (`num_points` - 1) predecessors to each observation are needed, the [`SeriesCache`] provided must have
+/// `num_leading_points` >= `num_points` - 1.
```

Comment:
Minor rewording and I added a section explaining `DataCache`, but maybe it should be moved close to the struct definition? It might be also important to point out that the length of the returned inner vectors is equal to (`cache.data.len() - cache.num_leading_points - cache.num_trailing_points`)?
```suggestion
/// Apply [`flatline_check`] to a whole [`DataCache`]
///
/// Each row of `DataCache.data` can be represented as follows:
/// |---|----------|---|
/// where the first and last sections are `DataCache.num_leading_points` and
/// `DataCache.num_trailing_points` long, respectively.
/// The actual observations to be QCed (i.e. flagged) lie in the middle section.
///
/// For each row in `DataCache.data`:
/// - If `num_points` observations are identical, the last one will be flagged as [`Flag::Fail`].
/// - If any of the last `num_points` observations are missing, it will be flagged as [`Flag::DataMissing`].
/// - Otherwise they will be flagged as [`Flag::Pass`].
///
/// As (`num_points` - 1) predecessors to each observation are needed, the [`DataCache`] provided must have
/// `num_leading_points` >= `num_points` - 1.
```

Reply:
fixes applied in person with Manuel